File: drag-drop.cs
const draggables = document.querySelectorAll("img[draggable='true']");
const slots = document.querySelectorAll(".slot");

draggables.forEach(img => {
    img.addEventListener("dragstart", event => {
        event.dataTransfer.setData("text", event.target.id);
    });
});

slots.forEach(slot => {
    slot.addEventListener("dragover", event => {
        event.preventDefault();
    });

    slot.addEventListener("drop", event => {
        const draggedId = event.dataTransfer.getData("text");
        const correctMatch = slot.getAttribute("data-match");

        if (draggedId === correctMatch) {
            const draggedElement = document.getElementById(draggedId);
            slot.appendChild(draggedElement);
            draggedElement.removeAttribute("draggable"); // Prevent further dragging
        }

        // Check if all slots are filled correctly
        if ([...slots].every(slot => slot.children.length > 0)) {
            window.location.href = "success.html"; // Redirect to Success page
        }
    });
});

File: drag-drop.js
const draggables = document.querySelectorAll("img[draggable='true']");
const dropZones = document.querySelectorAll(".drop-zone");

// Add drag animation
const addDragAnimation = (el) => {
    el.classList.add("dragging");
};

const removeDragAnimation = (el) => {
    el.classList.remove("dragging");
};

draggables.forEach(img => {
    img.addEventListener("dragstart", event => {
        event.dataTransfer.setData("text", event.target.id);
        addDragAnimation(event.target);
    });

    img.addEventListener("dragend", event => {
        removeDragAnimation(event.target);
    });
});

dropZones.forEach(zone => {
    zone.addEventListener("dragover", event => {
        event.preventDefault();
    });

    zone.addEventListener("drop", event => {
        event.preventDefault();
        const draggedId = event.dataTransfer.getData("text");
        const correctMatch = zone.getAttribute("data-match");
        const draggedElement = document.getElementById(draggedId);

        // Prevent multiple items in one zone
        if (zone.children.length === 0) {
            zone.appendChild(draggedElement);
            draggedElement.removeAttribute("draggable");
        }

        // Check if all zones are correctly matched
        const allCorrect = [...dropZones].every(zone => {
            const child = zone.querySelector("img");
            return child && child.id === zone.getAttribute("data-match");
        });

        if (allCorrect) {
            // Show success message or redirect
            alert("Success! All items matched correctly.");
            // window.location.href = "success.html";
        }
    });
});
